tests/typing_helpers.py

```
@@ -0,0 +1,202 @@
+def constant(value: T) -> Callable[..., T]:
+    def _inner(*_args: object, **_kwargs: object) -> T:
+        return value
+
+    return _inner
+
+
+def approx(
+    expected: object,
+    *,
+    rel: object | None = None,
+    abs: object | None = None,
```

Ruff (0.15.14)

[error] Function argument `abs` is shadowing a Python builtin

(A002)